src/main/java/com/google/api/codegen/transformer/DiscoTypeTable.java

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.discovery.Schema;
+import com.google.api.codegen.util.TypeAlias;
+import com.google.api.codegen.util.TypeTable;
+import java.util.Map;
+
+/**
+ * A DiscoTypeTable manages the imports for a set of fully-qualified type names, and provides helper
+ * methods for importing instances of TypeRef.
+ */
+public class DiscoTypeTable {
```

Comment:
Refactoring to SchemaTypeTable for now. I'm not sure right now if or how this class can be reused for tracking Discovery methods or resources too.